packages/@aws-cdk/aws-lambda-nodejs/lib/function.ts

```diff
@@ -0,0 +1,140 @@
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as cdk from '@aws-cdk/core';
+import * as crypto from 'crypto';
+import * as fs from 'fs';
+import * as path from 'path';
+import { build } from './build';
+import { nodeMajorVersion, parseStackTrace } from './util';
+
+/**
+ * Properties for a NodejsFunction
+ */
+export interface NodejsFunctionProps extends lambda.FunctionOptions {
+  /**
+   * Path to the entry file (JavaScript or TypeScript).
+   *
+   * @default - Derived from the name of the defining file and the construct's id.
+   * If the `NodejsFunction` is defined in `stack.ts` with `my-handler` as id
+   * (`new NodejsFunction(this, 'my-handler')`), the construct will look at `stack.my-handler.ts`
+   * and `stack.my-handler.js`.
+   */
+  readonly entry?: string;
+
+  /**
+   * The name of the exported handler in the entry file.
+   *
+   * @default handler
+   */
+  readonly handler?: string;
+
+  /**
+   * The runtime environment. Only runtimes of the Node.js family are
+   * supported.
+   *
+   * @default Runtime.NODEJS_12_X
+   */
+  readonly runtime?: lambda.Runtime;
+
+  /**
+   * Whether to minify files when bundling.
+   *
+   * @default false
+   */
+  readonly minify?: boolean;
+
+  /**
+   * Whether to include source maps when bundling.
+   *
+   * @default false
+   */
+  readonly sourceMaps?: boolean;
+
+  /**
+   * The build directory
+   *
+   * @default - `.build` in the entry file directory
+   */
```

Comment:
Feels like this should be derived from the entry point file name in case there are multiple handlers in the same directory (now that it’s possible)

Reply:
The refers to the "main" build directory, in there subfolders (with name derived from the entry point file name) are created

See https://github.com/jogold/aws-cdk/blob/1f3e639ec71517f8bd6b03b432a313363d87326d/packages/%40aws-cdk/aws-lambda-nodejs/lib/function.ts#L78-L79

Reply:
ok, cool